### ISSUE TITLE
how to use catalogue for components that require state

### ISSUE THREAD

**Issue by olivermt** [surface-ui/surface_bulma#8]
If not solved yet, could we have a discussion on how to do stuff like Modal where you pass it a show true/false.

To use a modal (from this lib) you need a liveview or a live component that controls it.
How do we set up demos in surface catalogue for componets like that?

**Comment by msaraiva** [comment]
Hey @olivermt!

Examples and Playgrounds are liveviews, i.e. they can hold state. All you have to do is to initialize `data props`and pass it to the `subject`, in your case, the Dialog. Then you can manipulate `props` as much as you like. The tool will update the component accordingly. I don't have the code of your dialog but here's an example of a fake one so you can see how it works:

```elixir
defmodule SurfaceBulma.FakeDialog do
  use Surface.Component

  prop show, :boolean
  prop message, :string

  def render(assigns) do
    ~H"""
    <div
      style="background: white; border: 1px solid; padding: 1.5rem; margin: 1.5rem; text-align: center;"
      :show={{ @show }}>
      {{ @message }}
    </div>
    """
  end
end

defmodule SurfaceBulma.FakeDialog.Playground do
  use Surface.Catalogue.Playground,
    subject: SurfaceBulma.FakeDialog,
    catalogue: SurfaceBulma.Catalogue,
    height: "200px"

  alias SurfaceBulma.Button

  data props, :map, default: %{show: false, message: "A fake message"}

  def render(assigns) do
    ~H"""
    <Button label="Open" color="info" click="show" disabled={{ @props.show }}/>
    <Button label="Close" color="danger" click="hide" disabled={{ !@props.show }}/>
    <!-- This is the subject of this Playground. It must receive @props  -->
    <FakeDialog :props={{ @props }}/>
    """
  end

  def handle_event("show", _, socket) do
    {:noreply, update(socket, :props, &%{&1 | show: true})}
  end

  def handle_event("hide", _, socket) do
    {:noreply, update(socket, :props, &%{&1 | show: false})}
  end
end
```

You should see something like this:

![image](https://user-images.githubusercontent.com/457237/106600182-cf92a400-6538-11eb-9b22-4307cd80bded.png)

**Comment by olivermt** [comment]
Fantastic!

**Comment by msaraiva** [comment]
@olivermt just in case you need some fast feedback, feel free to create a PR as soon as you have any of the new components + examples/playground ready. I'll be glad to review it. 

BTW, I don't think you need to create examples/playgrounds for all components before releasing `surface_bulma`. We can add them incrementally in later patch versions.

**Comment by olivermt** [comment]
I didn't even plan to do a single one before releasing :D 

A bit off topic for this issue, but my plan before releasing is:
- Modal
- Card, Card.Header, Card.Footer
- Add a sortable prop to Column in Table that materializes as clickable icon
- Add a sort_by prop to Table

Then I'll release 0.1.0. If I am not done with more than the modals/cards by thursday I'll clean up and release on friday regardless :)

**Comment by msaraiva** [comment]
Nice! One question. Have you tried the latest Bulma `v0.9.1`? Surface's website is still on `v0.8.2`, but feel free to move on and update it if you like. Last time I moved from `0.7` to `0.8` was straightforward. The only thing I noticed was a couple of minor differences regarding colors and shadows. The components behaviour didn't change at all. In the worst case, I can still point to an earlier version of the lib for a while.
